src/components/Input/WalletInput.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, useRef, MutableRefObject } from 'react';
+import Blockies from 'react-blockies';
+import {
+    DivStyledBlockie,
+    DivWrapperStyled,
+    InputStyled,
+    LabelStyled,
+    StrongStyled,
+    DivStyledEllipsis,
+} from './Input.styles';
+import type { WalletInputProps } from './types';
+import { getEllipsisTxt } from '../../web3utils';
+
+const WalletInput: React.FC<WalletInputProps> = ({
```

Comment:
You have done a great job here, but if I am honest I don't love this solution, sorry man I hate to give feedback when its not 100% good, and lets focus on the positive here you have done a great work and learned the kit more. 

The reason I suggested to import `Input` is because now we have two versions of `Input` to maintain. Two prop interfaces, logic and markup. If another dev or two (or more) solved their Input problem like this we would have a mess on our hands. As this is not a critical component for now. Lets park this task and come back to it. If I am honest I'm not very happy with `Input` right now and I might give it some love in the next few days. Then I was thinking to set this as a goal you can work on and we can tick off for your career progress at Moralis. How does that sound? Sorry to be the barer of not good news